### PR TITLE
feat(medusa, core-flows): add retrieve stock location endpoint to api-v2

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: integration-tests/modules/__tests__/stock-location/index.spec.ts [hunk 1]
```diff
@@ -52,5 +52,46 @@ medusaIntegrationTestRunner({
...
+
+        expect(response.status).toEqual(200)
+        expect(response.data.stock_location).toEqual(
+          expect.objectContaining({ id: locationId, ...location })
+        )
+      })
+
+      it("should get a stock location", async () => {
+        let error
+        await api
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/stock-location/index.spec.ts]: comment: Same comment as seen on other PR. Rest API tests should live in `integration-tests/api`. 

Let's move forward with this PR and migrate later.

File: packages/medusa/src/api-v2/admin/stock-locations/[id]/route.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+} from "@medusajs/utils"
+import { MedusaRequest, MedusaResponse } from "../../../../types/routing"
+
+import { MedusaError } from "@medusajs/utils"
+
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const { id } = req.params
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const [stock_location] = await remoteQuery(
```
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/stock-locations/[id]/route.ts:14]: comment: @carlos-r-l-rodrigues, your idea of adding a "throw-if-not-found" mechanism to remote query could make sense. I think in most endpoints, we are not checking for the existence of the retrieved resource when doing a get by ID.

**Review by olivermrbl — APPROVED** [review]

File: packages/medusa/src/api-v2/admin/stock-locations/[id]/route.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/api-v2/admin/stock-locations/[id]/route.ts:14]: Or alternatively, we add the `if (!result) {` to all get one endpoints, but I think having this baked in would be nice.

**Review by olivermrbl — APPROVED** [review]

**Comment by olivermrbl** [comment]
@pKorsholm, can you rebase and change base branch to develop before merging?

**Pull request merged**